src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        size++;
+        checkIndexInBounds(index);
+        System.arraycopy(elementData, index, elementData, index + 1, size - index - 1);
+        elementData[index] = value;
```

Comment:
size shouldn't be incremented before the index check. Better place line 27 at it the very end of the method